tests/system/test_aggregation_query.py

```diff
@@ -93,6 +93,26 @@ def test_aggregation_query_with_alias(aggregation_query_client, nested_query):
         assert r.value > 0
 
 
+def test_aggregation_query_with_limit(aggregation_query_client, nested_query):
+    query = nested_query
+
+    aggregation_query = aggregation_query_client.aggregation_query(query)
+    aggregation_query.count(alias="total")
+    result = _do_fetch(aggregation_query)
+    assert len(result) == 1
+    for r in result[0]:
+        assert r.alias == "total"
+        assert r.value > 0
+
+    aggregation_query = aggregation_query_client.aggregation_query(query)
+    aggregation_query.count(alias="total_up_to", limit=2)  # count with limit = 1
```

Comment:
can we add a test where there are more results than the limit, but the returned count == limit?

Reply:
Yup, I've adjusted the test. Above, the query was called without `limit` and it returned count of 8.
After, it was called with `limit=2`, and it returned count of 2.